src/classes/UTIL_Describe.cls

```diff
@@ -370,4 +370,108 @@ public class UTIL_Describe {
         }
         return strFields.substring(0, strFields.length()-1);
     }
+
+    /*******************************************************************************************************
+    * @description Cache for the record type DeveloperName-to-ID map.
+    */
+    private static Map<String, Id> mapRecordTypes;
+    
+    /*******************************************************************************************************
+     * @description Returns a map of active, user-available RecordType IDs for a given SObjectType,
+     * keyed by each RecordType's unique, unchanging DeveloperName. 
+     * @param objectName Name of the object to retrieve record types for.
+     * @return Map<String, Id> Map of DeveloperName-to-ID of available record types for object specified. 
+     */
+    public static Map<String, Id> getRecTypesMapByDevName(String objectName) {	    
+	    return getRecTypesMap(objectName, true);
+	}
+	
+	/*******************************************************************************************************
+     * @description Returns a map of active, user-available RecordType IDs for a given SObjectType,
+     * keyed by each RecordType's Name. 
+     * @param objectName Name of the object to retrieve record types for.
+     * @return Map<String, Id> Map of Name-to-ID of available record types for object specified. 
+     */
+    @AuraEnabled
+    public static Map<String, Id> getRecTypesMapByName(String objectName) {
+        return getRecTypesMap(objectName, false);
+    }
+    
+    /*******************************************************************************************************
+     * @description Returns a map of active, user-available RecordType IDs for a given SObjectType, keyed
+     * by either the RecordType's Name or the Developer Name, depending on the value of the second param. 
+     * @param objectName Name of the object to retrieve record types for.
+     * @param devName True if we want the key of the returned map to be the record type's developer name.
+     * False if we want it to be the friendly name.
+     * @return Map<String, Id> Map of Name-to-ID or DeveloperName-to-ID (depending on the second paramater), 
+     * of available record types for object specified. 
+     */
+    private static Map<String, Id> getRecTypesMap(String objectName, Boolean devName) {      
+        if(mapRecordTypes == null) {
+            mapRecordTypes = new Map<String, Id>();
+        
+            // Get the Describe Result
+            fillMapsForObject(objectName);
+            Schema.DescribeSObjectResult obj = objectDescribes.get(objectName);
+        
+            // Obtain ALL Active Record Types for the given SObjectType objectType
+            String soql = 'SELECT Id, Name, DeveloperName FROM RecordType ' + 'WHERE SObjectType = \'' 
```

Comment:
Would this support retrieval of the recordtypes for a managed object? I assume not...